### DISCUSSION TITLE
via API, my bot is able to reply to a post having `viewer.replyDisabled` set to `true`

### DISCUSSION THREAD

**Discussion by boly38** [bluesky-social/atproto#2576] [Clarifications and "Why?"] [answered]
Hi here 👋 

I'm owner of a bluesky bot who is searching some bluesky posts to reply to some post.

I encounter an issue on some post because the bot was able to post a reply for a post having `viewer.replyDisabled` flag set to `true`
this is described in details here : 
https://github.com/boly38/botEnSky/issues/41

I think I will fix that making the bot to simply ignore posts having this flag set and set to true.

But I would like to know if that kind of behavior is expected (bs.agent.post response is a success) from the backend. 
I would expect "Forbidden/unauthorized/bad request" or something like this for a user that is trying to reply (even via API) to a post that prevent reply.
Maybe it's simply the "client" responsability to deal with that flag IDK.

Thanks for your answer
Regards

**Comment by yamarten** [comment, marked as answer]
That's the right behavior in the specification. This behavior occurs because PDS, the server that accepts posts, and appview, which is a server that judges the availability of replies, are separate.

You can see only one endpoint, but in fact, feed queries are forwarded behind the scenes.

PDS is in principle ignorant of service logic. This has the advantage of being able to provide data for various services without updating the PDS each time. In return, at the timing of posting the data, the only possible verification is by the schema.

Also, if a new appview appears in the future, there is no guarantee that the value of `replyDisabled` is the same.

I think it's essentially the client's role to choose which appview rule to refer to. However, there is currently only one public bsky appview, so if you do not explicitly select appview, it will be selected automatically.
